src/app/src/components/annotations/annotator.tsx

```diff
@@ -270,6 +273,8 @@ export default class Annotator extends Component<
     this.filterAnnotationVisibility = this.filterAnnotationVisibility.bind(
       this
     );
+    this.setAnnotationOutline = this.setAnnotationOutline.bind(this);
```

Comment:
@virtuoso98 considering that the state is a single object that describes the annotation aesthetics, can you combine them into `this.setAnnotationOptions` and process this option by taking in both variables?

Reply:
I have done this based on your tips:
`  /**
   * Updates the annotationOptions, handling both boolean
   * and number case
   * @param {boolean | number } newOption - updatedOptions
   */
  private setAnnotationOptions(newOption: boolean | number): void {
    this.setState(
      prevState => {
        const config = prevState.annotationOptions;
        switch (typeof newOption) {
          case "boolean":
            config.isOutlined = newOption
              ? true
              : (!prevState.annotationOptions.isOutlined as any);
            break;
          case "number":
            config.opacity = newOption;
            break;
          default:
            break;
        }
        return { annotationOptions: config };
      },
      () => this.filterAnnotationVisibility()
    );
  }`